DLLModsSource/SonicGenerations/SonicUnleashedHUD/UnleashedHUD/HudTitle.cpp

```diff
@@ -0,0 +1,159 @@
+Chao::CSD::RCPtr<Chao::CSD::CProject> rcTitleScreen;
+Chao::CSD::RCPtr<Chao::CSD::CScene> rcTitleLogo_1, rcTitlebg, rcTitleMenu, rcTitleMenuScroll, rcTitleMenuTXT;
+boost::shared_ptr<Sonic::CGameObjectCSD> spTitleScreen;
+int currentTitleIndex = 0;
+
+void HudTitle::CreateScreen(Sonic::CGameObject* pParentGameObject)
+{
+	if (rcTitleScreen && !spTitleScreen)
+		pParentGameObject->m_pMember->m_pGameDocument->AddGameObject(spTitleScreen = boost::make_shared<Sonic::CGameObjectCSD>(rcTitleScreen, 0.5f, "HUD", false), "main", pParentGameObject);
+}
+
+void HudTitle::KillScreen()
+{
+	if (spTitleScreen)
+	{
+		spTitleScreen->SendMessage(spTitleScreen->m_ActorID, boost::make_shared<Sonic::Message::MsgKill>());
+		spTitleScreen = nullptr;
+	}
+}
+
+void HudTitle::ToggleScreen(const bool visible, Sonic::CGameObject* pParentGameObject)
+{
+	if (visible)
+		CreateScreen(pParentGameObject);
+	else
+		KillScreen();
+}
+
+void HudTitle::FreezeMotion(Chao::CSD::CScene* pScene)
+{
+	pScene->SetMotionFrame(pScene->m_MotionEndFrame);
+	pScene->m_MotionSpeed = 0.0f;
+	pScene->m_MotionRepeatType = Chao::CSD::eMotionRepeatType_PlayOnce;
+}
+
+void HudTitle::PlayAnimation(Chao::CSD::CScene* pScene, const char* name, Chao::CSD::EMotionRepeatType repeatType, float motionSpeed, float startFrame)
+{
+	pScene->SetMotion(name);
+	pScene->m_MotionRepeatType = repeatType;
+	pScene->m_MotionDisableFlag = 0;
+	pScene->m_MotionSpeed = motionSpeed;
+	pScene->SetMotionFrame(startFrame);
+	pScene->Update(0);
+}
+
+void HudTitle::IntroAnim(Chao::CSD::RCPtr<Chao::CSD::CScene> scene)
+{
+	scene->SetMotion("Intro_Anim");
+	scene->SetMotionFrame(0.0f);
+	scene->m_MotionSpeed = 1;
+	scene->m_MotionRepeatType = Chao::CSD::eMotionRepeatType_PlayOnce;
+	scene->Update(0.0f);
+}
+
+void UnleashedTitleText()
+{
+	rcTitleMenuTXT->SetPosition(0, 0);
+
+	if (currentTitleIndex < 0)
+		currentTitleIndex = 4;
+	else if(currentTitleIndex > 4)
```

Comment:
Put a space after `if`.

Reply:
You mean an extra space after the else if to fit with the if?

Reply:
Yes.